tests/test_type/test_bit.py

```diff
@@ -262,6 +263,12 @@ def definition(io):
     assert check_files_equal(__file__, f"build/TestBitInvert.v",
                              f"gold/TestBitInvert.v")
 
+    sim = PythonSimulator(TestInvert)
```

Comment:
Looks like this (and below stuff) implies that the python simulator is part of the main magma spec; meaning if the simulator is broken, magma is broken. That is ok - just making sure that is a conscious decision.

Reply:
Yea, that's the reason I added the test (so in future changes we would have caught errors related to these code paths)